### PR TITLE
Sync NetworkController getEIP1559Compatibility tests w/ core

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

This makes it easier to visually compare differences in the NetworkController unit tests between core and this repo.

Progresses https://github.com/MetaMask/core/issues/1197.

For reference, see core: https://github.com/MetaMask/core/blob/593e74c074d68587f99bd960901a3c6042e79d0b/packages/network-controller/tests/NetworkController.test.ts#L2849

## Manual Testing Steps

There should be no functional changes. All tests should pass.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by mcmire** [comment]
This PR is blocked by https://github.com/MetaMask/core/pull/1410

File: app/scripts/controllers/network/network-controller.test.ts [hunk 1]
```diff
@@ -723,204 +723,265 @@ describe('NetworkController', () => {
...
+              const isEIP1559Compatible =
+                await controller.getEIP1559Compatibility();
+
+              expect(isEIP1559Compatible).toBe(false);
+            });
+          });
         });
 
-        await controller.getEIP1559Compatibility();
-        await controller.getEIP1559Compatibility();
```
Comment (digiwand) [inline app/scripts/controllers/network/network-controller.test.ts:917]: Q: what was the thought behind removing this test?

Comment (mcmire) [reply app/scripts/controllers/network/network-controller.test.ts:917]: Good spot. My thought here was that this behavior is already being tested by two other kinds of tests elsewhere:

1. If `controller.store.getState().networkDetails.EIPS[1559]` is undefined, a call to `getEIP1559Compatibility` should change `networkDetails.EIPS[1559]` in state to either true or false (depending on whether the latest block has a `baseFeePerGas` property
2. If `networkDetails.EIPS[1559]` in state is already true or false, a call to `getEIP1559Compatibility` should not make a network request or make any changes to `networkDetails.EIPS[1559]` in state, but should just return that property

If you put these two together, then you get what this test was doing. That said, I can understand that this is more explicit in demonstrating that `getEIP1559Compatibility` essentially memoizes its result, so what do you think?

File: app/scripts/controllers/network/network-controller.test.ts [hunk 2]
```diff
@@ -723,204 +723,265 @@ describe('NetworkController', () => {
...
 
-          expect(supportsEIP1559).toBeTruthy();
+          expect(isEIP1559Compatible).toBe(false);
         });
       });
     });
 
-    describe('when the latest block does not have a baseFeePerGas property', () => {
-      it('stores the fact that the network does not support EIP-1559', async () => {
+    describe('if a provider has been set but networkDetails.EIPS in state already has a "1559" property', () => {
```
Comment (Gudahtt) [inline app/scripts/controllers/network/network-controller.test.ts:751]: Hmm. These tests seem incorrect; the provider isn't being set.

Comment (mcmire) [reply app/scripts/controllers/network/network-controller.test.ts:751]: PR created on the core side to fix these tests: https://github.com/MetaMask/core/pull/1420

Comment (mcmire) [reply app/scripts/controllers/network/network-controller.test.ts:750]: Tests fixed here: [cb7aafe](https://github.com/MetaMask/metamask-extension/pull/19419/commits/cb7aafefc1d0b6cc0b237999c29fd729622c4c52)

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]

**Pull request merged**